### PR TITLE
Creating a data model

### PR DESCRIPTION
**Pull request by Abdulwhbhussain**
[Project requirements](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/creating_data_model_project_v1_2.md#project-requirements)
Create a new app. Give it a meaningful name.
Initialize your project with Git.
Make sure that your project has Postgres database set up.
Build your project schema.
Create and run the necessary migration files.
Table and column names should match the ERD diagram.
note: photo for users table should be a link to an image
Foreign keys should be included.
Pay attention that in posts table the column holding the foreign key to the users table should be named author_id
All columns that are foreign keys should have a corresponding index.
Be sure to reference the from the sneak peek if you have any questions about table or column names or data relationships.

Any Suggestions will be welcomed.

### CODE REVIEW

**Comment by topeogunleye** [comment]
@Abdulwhbhussain, Hi, I am your code reviewer, Can you please quickly update your README file to use the Microverse template.

**Comment by topeogunleye** [comment]
Hi @Abdulwhbhussain, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that does not have a README file that uses the Microverse template.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @topeogunleye in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: db/migrate/20231024174016_create_posts.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+class CreatePosts < ActiveRecord::Migration[7.1]
+  def change
+    create_table :posts do |t|
+      t.references :user, null: false, foreign_key: true, index: true
```
Comment (codepantha) [inline db/migrate/20231024174016_create_posts.rb:4]: - [ ] According to the ERD diagram, ensure that the `user_id` column is called `author_id`. This is easily achieved by doing something like this 👇 
```ruby
t.references :author, foreign_key: { to_table: 'users' }
#### ERD Diagram
![image](https://github.com/Abdulwhbhussain/blog-app/assets/46839548/3b5c3f03-b9ec-4fd2-9eb8-f79c85fad83e)

```

**Review by codepantha — CHANGES_REQUESTED** [review]
# STATUS: CHANGES REQUIRED ♻️ 

## Hi @Abdulwhbhussain 👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights 🎯 
✔️ Descriptive PR
✔️ Created migration files to setup database tables 📌 
✔️ Good commit history
✔️ No linter errors

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- n/a

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: db/migrate/20231024174016_create_posts.rb (hunk 1 above)
Comment (Abdulwhbhussain) [reply db/migrate/20231024174016_create_posts.rb:4]: class RenameUserIdColumnToAuthorIdColumn < ActiveRecord::Migration[7.1]
  def change
    rename_column :posts, :user_id, :author_id
  end
end

"I did it using this migration code and file."

Comment (Abdulwhbhussain) [reply db/migrate/20231024174016_create_posts.rb:4]: This is not right I used **rename_column** method instead of doing your way.

**Review by Felix45 — APPROVED** [review]
# Approved :green_circle: 
Hi @Abdulwhbhussain, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights :heavy_check_mark: 
- [x] All migrations are set up correctly :white_check_mark: 
- [x] Application functions as expected :white_check_mark: 
- [x] Git flow is used correctly :white_check_mark: 
- [x] Pull request has a descriptive title :white_check_mark: 
- [x] Repository has a professional readme file :white_check_mark:   

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**